ios/FluentUI/Bottom Sheet/BottomCommandingController.swift

```diff
@@ -0,0 +1,596 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// Persistent commanding surface displayed at the bottom of the available area.
+///
+/// The presentation style automatically varies depending on the current horizontal `UIUserInterfaceSizeClass`:
+///
+/// `.unspecified` and `.compact` - the surface is displayed as an expandable bottom sheet.
+///
+/// `.regular` -  the surface is displayed as a floating bottom bar.
+///
+/// In both styles, `heroItems` are always presented in a horizontal stack.
+/// Items from the `expandedListSections` are either presented in an expanded sheet or a popover, depending on the current style.
+///
+@objc(MSFBottomCommandingController)
+open class BottomCommandingController: UIViewController {
+
+    /// Items to be displayed in an area that's always visible. This is either the top of the the sheet,
+    /// or the main bottom bar area, depending on current horizontal UIUserInterfaceSizeClass.
+    ///
+    /// Only the first 5 items will be retained and displayed.
+    @objc open var heroItems: [CommandingItem] = [] {
+        willSet {
+            clearAllItemViews(in: .heroSet)
+        }
+        didSet {
+            if heroItems.count > 5 {
+                assertionFailure("At most 5 hero commands are supported. Only the first 5 items will be used.")
```

Comment:
Should avoid assertionFailure since it actually hardcodes the full file path of the compilation unit in the outputted binary. preconditionFailure should be used instead